pythonwhat/probe.py

```diff
@@ -0,0 +1,218 @@
+import pprint as pp
+import itertools
+import inspect
+from functools import partial
+from collections import OrderedDict
+
+TEST_NAMES = [
+    "test_mc",
+    "test_or",
+    "test_with",
+    "test_list_comp",
+    "test_dict_comp",
+    "test_generator_exp",
+    "test_import",
+    "test_object",
+    "test_correct",
+    "test_if_else",
+    "test_if_exp",
+    "test_for_loop",
+    "test_function",
+    "test_print",
+    "test_function_v2",
+    "test_operator",
+    "test_try_except",
+    "test_data_frame",
+    "test_dictionary",
+    "test_while_loop",
+    "test_student_typed",
+    "test_object_accessed",
+    "test_output_contains",
+    "test_lambda_function",
+    "test_expression_result",
+    "test_expression_output",
+    "test_function_definition",
+    "test_object_after_expression"
+]
+
+SUB_TESTS = {
```

Comment:
I see that in line 157, you put as TODO that you want to make this automatic. How hard is this?